src/mudlet.cpp

```diff
@@ -4593,26 +4526,107 @@ void mudlet::setNetworkRequestDefaults(const QUrl& url, QNetworkRequest& request
-        // the tab to match - without triggering slot_tabChanged(int)
-        int tabToBeActive = mpTabBar->tabIndex(newActiveHostName);
-        if (tabToBeActive >= 0) {
-            mpTabBar->blockSignals(true);
-            mpTabBar->setCurrentIndex(tabToBeActive);
-            mpTabBar->blockSignals(false);
+    int newActiveTabIndex = hostNameToTabMap.value(newActiveHostName, -1);
+
+    if (mpCurrentActiveHost && mpCurrentActiveHost->mpConsole) {
+        // Tell the old profile that it is losing focus:
+        TEvent focusLostEvent {};
+        focusLostEvent.mArgumentList << QLatin1String("sysProfileFocusChangeEvent");
+        // Boolean arguments are carried as "0" for false or "1" for true,
+        // This is for the profile that is losing focus:
+        focusLostEvent.mArgumentList << QLatin1String("0");
+        focusLostEvent.mArgumentTypeList << ARGUMENT_TYPE_STRING << ARGUMENT_TYPE_BOOLEAN;
+        mpCurrentActiveHost->raiseEvent(focusLostEvent);
+
+        if (!mMultiView) {
+            // We only have to hide the current profile under the tab if NOT
+            // in multi-view mode:
+            mpCurrentActiveHost->mpConsole->hide();
         }
-        mpCurrentActiveHost = pHost;
-        updateDiscordNamedIcon();
-        dactionInputLine->setChecked(mpCurrentActiveHost->getCompactInputLine());
-        pHost->updateDisplayDimensions();
-        emit signal_profileActivated(pHost, tabToBeActive);
     }
+
+    // Need to change to the right tab without triggering slot_tabChanged(int)
+    // as that might even be the caller of this method!
+    if (newActiveTabIndex >= 0) {
+        mpTabBar->blockSignals(true);
+        mpTabBar->setCurrentIndex(newActiveTabIndex);
+        mpTabBar->blockSignals(false);
+    }
+
+    mpCurrentActiveHost = pHost;
+    mpCurrentActiveHost->mpConsole->show();
+    mpCurrentActiveHost->mpConsole->repaint();
+    mpCurrentActiveHost->mpConsole->refresh();
+    mpCurrentActiveHost->mpConsole->mpCommandLine->repaint();
+    // CHECKME: Although we no longer force the focus to the profiles main
+    // console's command line here (because it isn't always appropriate) we are
+    // redrawing the main main console AND it's primary command line. That
+    // being the case should we perhaps also be repainting any "mini consoles"
+    // and/or any additional command lines, that have either been added to the
+    // main console or enabled on any mini console or "user window"?
+
+    // Regenerate the multi-view mode if it is enabled:
+    reshowRequiredMainConsoles();
+
+    // Reset the styles to reflect those of the now active profile:
+    mpMainToolBar->setStyleSheet(mpCurrentActiveHost->mProfileStyleSheet);
+    mpTabBar->setStyleSheet(mpCurrentActiveHost->mProfileStyleSheet);
+    menuBar()->setStyleSheet(mpCurrentActiveHost->mProfileStyleSheet);
+
+    // Tell the old profile that it is losing focus via a Mudlet event:
```

Comment:
Oops! A bit of a copy-pasta error:
```suggestion
    // Tell the new profile that it is gaining focus via a Mudlet event:
```